app/controllers/adhoq/executions_controller.rb

```diff
@@ -23,7 +23,11 @@ def current_query
     end
 
     def respond_report(report)
-      send_data report.data, type: report.mime_type, filename: report.name, disposition: 'attachment'
+      if Adhoq.current_storage.direct_download?
```

Comment:
`report` should know its own storage and we should ask it whether direct_download supproted, like l12 `report#on_the_fly?`.

Hmm, but I noticed that `Report#data` already have same issue. Improve it in future.